src/synapse.ts

```diff
@@ -120,13 +123,23 @@ export class Synapse {
       )
     }
 
+    // Determine Pandora address early for discovery initialization
+    const pandoraAddress = options.pandoraAddress ?? CONTRACT_ADDRESSES.PANDORA_SERVICE[network]
+    if (pandoraAddress === '' || pandoraAddress === undefined) {
+      throw new Error(`No Pandora service address configured for network: ${network}`)
+    }
+
+    // Create default discovery if not provided
+    const pieceDiscovery = options.pieceDiscovery ?? new ChainDiscovery(provider, pandoraAddress)
```

Comment:
@rvagg is this the place where to plug in the CDN-specific downloads?

```suggestion
    // Create default discovery if not provided
    const pieceDiscovery = options.pieceDiscovery ?? 
      (options.withCDN ? new FilCdnDiscovery(network) : new ChainDiscovery(provider, pandoraAddress))
```